src/odemis/gui/comp/overlay/base.py

```diff
@@ -1205,35 +1694,73 @@ def update_projection(self, idx, b_pos, shiftscale):
             logging.debug("Updating view position of %s, shiftscale %s", idx, shiftscale)
             self.v_points[idx] = Vec(self.cnvs.buffer_to_view(b_pos))
 
+    def _calc_center(self):
+        """Calculate the center of selection."""
+        n = len(self.v_points)
+        sum_x = sum(p.x for p in self.v_points)
+        sum_y = sum(p.y for p in self.v_points)
+        centroid_x = sum_x / n
+        centroid_y = sum_y / n
```

Comment:
Nowadays, you can use the `statistics` module from the standard library:
`centroid_x = statistics.mean(p.x for p in self.v_points)`